preheaders/billing.php

```diff
@@ -158,6 +158,39 @@
         }
     }
 	
+	// Check reCAPTCHA if needed.
+	global $recaptcha, $recaptcha_validated;
+	if (  $recaptcha == 2 || ( $recaptcha == 1 && pmpro_isLevelFree( $pmpro_level ) ) ) {
+		global $recaptcha_privatekey;
+		if ( isset( $_POST["recaptcha_challenge_field"] ) ) {
+			//using older recaptcha lib
+			$resp = recaptcha_check_answer( $recaptcha_privatekey,
+				$_SERVER["REMOTE_ADDR"],
+				$_POST["recaptcha_challenge_field"],
+				$_POST["recaptcha_response_field"] );
+
+			$recaptcha_valid  = $resp->is_valid;
+			$recaptcha_errors = $resp->error;
+		} else {
+			//using newer recaptcha lib
+			// NOTE: In practice, we don't execute this code because
+			// we use AJAX to send the data back to the server and set the
+			// pmpro_recaptcha_validated session variable, which is checked
+			// earlier. We should remove/refactor this code.
+			$reCaptcha = new pmpro_ReCaptcha( $recaptcha_privatekey );
+			$resp      = $reCaptcha->verifyResponse( $_SERVER["REMOTE_ADDR"], $_POST["g-recaptcha-response"] );
```

Comment:
Do we need to sanitize any of the $_POST data before using/sending?

Reply:
I think we let reCAPTCHA deal with that.